becquerel/core/fitting.py

```diff
@@ -338,12 +340,45 @@ def __str__(self):
             '        x: {}\n'.format(self.x) +
             '        y: {}\n'.format(self.y) +
             '    y_unc: {}\n'.format(self.y_unc) +
+            '    xmode: {}\n'.format(self.xmode) +
+            '    ymode: {}\n'.format(self.ymode) +
             '       dx: {}\n'.format(self.dx) +
             '      roi: {}'.format(self.roi)
         )
 
     __repr__ = __str__
 
+    @classmethod
+    def from_spectrum(cls, model, spec, xmode, ymode, roi=None):
```

Comment:
I am sorry this wasn't clear earlier. I would not have this code here at all, and move it all to `Spectrum.fit`. While this works sort of, it is a easy source of circular dependencies (for example if we want to add a test that this really only can take `Spectrum` objects we would need to `import spectrum`. I don't see a use-case where this would be useful and `Spectrum.fit` can't accomplish the same thing.

Reply:
Makes sense. I'll fix that now.

Reply:
Resolving based on the latest commit